libcxx/test/std/algorithms/alg.sorting/alg.set.operations/set.intersection/set_intersection_complexity.pass.cpp

```diff
@@ -43,33 +43,31 @@
 
 #include "test_iterators.h"
 
-namespace {
-
-// __debug_less will perform an additional comparison in an assertion
-static constexpr unsigned std_less_comparison_count_multiplier() noexcept {
-#if _LIBCPP_HARDENING_MODE == _LIBCPP_HARDENING_MODE_DEBUG
-  return 2;
+// Debug mode provides no complexity guarantees, testing them would be a waste of effort.
+#ifdef _LIBCPP_HARDENING_MODE_DEBUG
```

Comment:
This is wrong. `_LIBCPP_HARDENING_MODE_DEBUG` is always defined.

Reply:
```suggestion
#if defined(_LIBCPP_HARDENING_MODE_DEBUG) && _LIBCPP_HARDENING_MODE_DEBUG
```

We still need to check for `if defined(_LIBCPP_HARDENING_MODE_DEBUG)` to avoid `-Wundef` when testing non-libc++ libraries.